q2_qemistree/_hierarchy.py

```diff
@@ -8,58 +8,87 @@
 
 import biom
 import pandas as pd
-from sklearn.metrics import pairwise_distances
-from scipy.spatial.distance import squareform
+from scipy.spatial.distance import squareform, pdist
 from scipy.cluster.hierarchy import linkage
 from skbio import TreeNode
+import numpy as np
 from q2_feature_table import merge
 
 from ._collate_fingerprint import collate_fingerprint
-from ._match import match_label
+from ._match import match_tables
 from ._semantics import CSIDirFmt
 
 
-def build_tree(relabeled_fingerprints: pd.DataFrame) -> TreeNode:
+def pdist_union(u, v):
+    b = np.double(np.bitwise_or(u, v).sum())
+
+    return b
+
+
+def pairwise_jaccard_modified(merged_fps: pd.DataFrame,
+                              merged_fdata: pd.DataFrame,
+                              mz_tolerance: float):
```

Comment:
Can you add the docstring for this function:

Reply:
Also, including a description of why we compute the distance this way would be very handy